diff --git a/mirac_network/mirac-broker.hpp b/mirac_network/mirac-broker.hpp
index 50dc1a9..7a1a554 100644
--- a/mirac_network/mirac-broker.hpp
+++ b/mirac_network/mirac-broker.hpp
@@ -35,7 +35,7 @@ class MiracBroker : public wds::Peer::Delegate
 {
     public:
         MiracBroker (const std::string& listen_port);
-        MiracBroker(const std::string& peer_address, const std::string& peer_port, uint timeout = 3000);
+        MiracBroker(const std::string& peer_address, const std::string& peer_port, uint timeout = 10000);
         virtual ~MiracBroker ();
         unsigned short get_host_port() const;
         std::string get_peer_address() const;
